agentless.tf

```diff
@@ -4,6 +4,8 @@
 //        --arn arn:aws:iam::${TF_VAR_ACCOUNT_ID}:role/chainguard-agentless \
 //        --group system:masters --username admin
 resource "aws_iam_role" "agentless_role" {
+  for_each = toset(local.enforce_group_ids)
```

Comment:
See comment below in canary-role. I think we need to keep one-role and have the trust relationship on that one rule include all Enforce IAM Groups

Reply:
Yeah, I was just looking into this and came across this PR.

I think what we need here is:
```yaml
"Condition" : {
  "StringEquals" : {
    "issuer.${var.enforce_domain_name}:aud" : "amazon"
  },
  "ForAnyValue:StringEquals" : {
    "issuer.${var.enforce_domain_name}:sub": [
      # This isn't valid, but intended to express intent.
      "agentless:{group}" for group in local.enforce_group_ids
    ]
  }
}
```

See here for "ForAnyValue" docs: https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_multi-value-conditions.html

Reply:
ack, fixed.